src/pdm/project/core.py

```diff
@@ -762,3 +763,34 @@ def is_distribution(self) -> bool:
             return settings["distribution"]
         else:
             return True
+
+    def get_setting(self, key: str) -> Any:
+        """
+        Get a setting from its dotted key (without the `tool.pdm` prefix).
+
+        Returns `None` if the key does not exists.
+        """
+        try:
+            return reduce(operator.getitem, key.split("."), self.pyproject.settings)
+        except tomlkit.exceptions.NonExistentKey:
+            return None
+
+    def env_or_setting(self, var: str, key: str) -> Any:
+        """
+        Get a value from environment variable and fallback on a given setting.
+
+        Returns `None` if both the environment variable and the key does not exists.
+        """
+        return os.getenv(var.upper()) or self.get_setting(key)
+
+    def env_or_setting_list(self, var: str, key: str) -> list[str]:
```

Comment:
This has too much assumption on the structure of the value. I prefer not to expose this as a utility method, but instead, parse the value every time being used, possibly with a helper function.

Reply:
Done: it's now a private helper on `BaseEnvironment`.